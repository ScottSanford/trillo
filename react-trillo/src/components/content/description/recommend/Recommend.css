.recommend {
    align-items: center;
    font-size: 1.3rem;
    color: var(--color-grey-dark-3);
    display: flex;
}

.recommend__count {
    margin-right: auto;
}

.recommend__friends {
    display: flex;
}

.recommend__photo {
    border: 3px solid var(--color-white);
    border-radius: 50%;
    box-sizing: content-box;
    height: 4rem;
    width: 4rem;
}   

.recommend__photo:not(:last-child) {
    margin-right: -2rem;
}
