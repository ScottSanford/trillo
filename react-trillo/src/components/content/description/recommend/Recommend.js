import React from 'react'
import './Recommend.css'

const friendRecommendList = [
  { img: 'img/user-3.jpg', name: 'Friend 1'},
  { img: 'img/user-4.jpg', name: 'Friend 2'},
  { img: 'img/user-5.jpg', name: 'Friend 3'},
  { img: 'img/user-6.jpg', name: 'Friend 4'},
]

export default function Recommend() {

  const recommendFriendImages = friendRecommendList.map(friend => {
    return (
      <img key={friend.name} src={friend.img} alt={friend.name} className="recommend__photo" />
    )
  })

  return (
    <div className="recommend">
      <p className="recommend__count">Lucy and 3 other friends recommend this hotel.</p>
      <div className="recommend__friends">
        {recommendFriendImages}
      </div>
    </div>
  )
}