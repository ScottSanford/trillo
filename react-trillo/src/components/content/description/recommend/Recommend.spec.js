import React from 'react'
import { shallow } from 'enzyme'
import Recommend from './Recommend'

describe('Recommend Component', () => {

	it('should render correctly with no props', () => {

		const wrapper = shallow(<Recommend />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should display a list of 2 images if 2 recommendations are passed in', () => {
		const data = [
			{ img: 'test1.png', name: 'test 1'},
			{ img: 'test2.png', name: 'test 2'},
		]

		const wrapper = shallow(<Recommend list={data} />)
		expect(wrapper.find('.recommend__friends').children()).toHaveLength(2)
	})

	it('should exclude an image from the recommended friends list if no image is available', () => {
		const data = [
			{ img: 'test1.png', name: 'test 1'},
			{ img: 'test2.png', name: 'test 2'},
			{ img: '', name: 'test 3'}
		]

		const wrapper = shallow(<Recommend list={data} />)
		expect(wrapper.find('.recommend__friends').children()).toHaveLength(2)
	})
})