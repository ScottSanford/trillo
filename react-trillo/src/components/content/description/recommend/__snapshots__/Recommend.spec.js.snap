// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommend Component should match the snapshot 1`] = `
<div
  className="recommend"
>
  <p
    className="recommend__count"
  >
    Lucy and 3 other friends recommend this hotel.
  </p>
  <div
    className="recommend__friends"
  />
</div>
`;
